OpenRA.Mods.Common/CampaignProgress.cs

```diff
@@ -0,0 +1,124 @@
+using System;
+using System.Collections.Generic;
+using System.IO;
+using OpenRA.FileSystem;
+using OpenRA.Graphics;
+
+namespace OpenRA.Mods.Common
+{
+	public class CampaignProgress
+	{
+		private static string progressFile = Platform.ResolvePath("^", "cnc-progress.yaml");
+		private static bool saveProgressFlag = false;
+        private static string playedMission = "";
+
+		public static void Init()
+		{
+			ModMetadata initialMod = null;
+			ModMetadata.AllMods.TryGetValue(Game.Settings.Game.Mod, out initialMod);
+			var mod = initialMod.Id;
+			progressFile = Platform.ResolvePath("^", mod + "-progress.yaml");
+		}
+
+        public static void SetPlayedMission(string mission)
+        {
+            playedMission = mission;
+        }
+
+		public static void SetSaveProgressFlag()
```

Comment:
This and the following two would be better represented by a single `SaveProcessFlag` property.